docs/troubleshooting/reverse-queries.rst

```diff
@@ -2,6 +2,9 @@
 Reverse Queries fail when the partition is larger than the maximum safe size
 =============================================================================
 
+.. note::
+  The problem described on this page may occur in ScyllaDB Enterprise 2021.1 or earlier and ScyllaDB Open Source 4.6 or earlier. In later versions, reverse queries that read too much data are aborted (see https://github.com/scylladb/scylla/issues/5804).
```

Comment:
> In later versions, reverse queries that read too much data are aborted (see https://github.com/scylladb/scylla/issues/5804).

This is incorrect. In later versions, reverse queries are not aborted anymore, as they have a new efficient implementation which does not need to reverse partitions in-memory. This is not 100% true however. In particular partition-range scans that go to the disk can still hit the memory limit.

Reply:
@denesb 

>  In later versions, reverse queries are not aborted anymore, as they have a new efficient implementation which does not need to reverse partitions in-memory. This is not 100% true however.

- Does that mean that, for example, a ScyllaDB user running version 2022.1 may still hit this problem (it's just less likely than e.g., on version 2020.1)?
- Do we still recommend avoiding reverse queries (_"It is highly recommended that when using the CQL ORDER BY clause, the order in the query is the same as the data model"_)?

If that's the case, we can leave the page unchanged because the user of any ScyllaDB version may find the information helpful.

Reply:
5.0/2022.1 is the version that contains the up-do-date native reverse read implementation. On these versions reverse queries should be well supported, I don't think we need to warn users to avoid them anymore. We should still strongly recommend that they use the correct ordering, so they have to read in reverse the least amount possible, but doing reverse reads is not an anti-pattern anymore.

Reply:
> Does that mean that, for example, a ScyllaDB user running version 2022.1 may still hit this problem (it's just less likely than e.g., on version 2020.1)?

Yes, they may still hit this as not 100% of reverse reads are covered with the new efficient implementation. But this should be rare now.

> Do we still recommend avoiding reverse queries ("It is highly recommended that when using the CQL ORDER BY clause, the order in the query is the same as the data model")?

See my previous comment. Yes we should keep this recommendation, but we no longer have to strongly recommend against reverse reads in general.